deploy-service/common/src/main/java/com/pinterest/deployservice/alerts/AlertAction.java

```diff
@@ -0,0 +1,16 @@
+package com.pinterest.deployservice.alerts;
+
+import com.pinterest.deployservice.bean.DeployBean;
+import com.pinterest.deployservice.bean.EnvironBean;
+
+
+/**
+ * AlertAction is the base class for an action for an alert
+ */
+public abstract class AlertAction {
+  public abstract Object perform(EnvironBean environ,
+                                 DeployBean lastDeploy,
```

Comment:
What happens if there were multiple deploys in the window?

Reply:
Rollback goes to the last known good deploy before the window so all the deploys in the window are rolled back. However, markbadbuild only marks the build of the latest deploy. It is a bit tricky

Reply:
I guess the problem I see is with auto deploy. If I want to combine rollback & mark build as bad I would really want to mark all builds deployed in the window as bad. Wouldn't auto-deploy otherwise pick a build in the window if there happen to be multiple?

Reply:
@smeder I see. This is an issue if (waiting to promote period <window size). I can make markbadbuild to label the build of the first deploy in the window.  In general, I think people configure a workflow like (deploy a build, waiting x minutes with no alerts, promote)  and x is larger than the window size in general.

Reply:
Ok - I think I can live within those constraints.